### PR TITLE
Update Dissolve algorithm

### LINKED ISSUE #3049: [needs-docs] Review the Dissolve algorithm description
Original commit: https://github.com/qgis/QGIS/commit/0396abfd01e5817701dec7d249eaee02d38ea236 by nyalldawson

* The current description focuses on geometries and assumes only line or vector layers can be dissolved
* Rename label to look more meaningful and closer to GDAL Dissolve's

### PR DESCRIPTION
**Pull request by DelazJ**
Fix #3049

### CODE REVIEW

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 1]
```diff
@@ -785,19 +785,19 @@ See also
 
 Dissolve
 --------
-Takes a polygon or line vector layer and combines their geometries into new
-geometries creating a new layer.
-
-One or more attributes can be specified to dissolve only geometries belonging to
-the same class (having the same value for the specified attributes), alternatively
-all geometries can be dissolved.
+Takes a vector layer and combines their features into new features.
```
Comment (havatv) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: their - its

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 2]
```diff
@@ -785,19 +785,19 @@ See also
...
+One or more attributes can be specified to dissolve features belonging to the
+same class (having the same value for the specified attributes), alternatively
+all features can be dissolved in a single one.
```
Comment (havatv) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: in a single one -> to a single feature

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 3]
```diff
@@ -785,19 +785,19 @@ See also
...
 All output geometries will be converted to multi geometries. In case the input is
 a polygon layer, common boundaries of adjacent polygons being dissolved will get
 erased.
 
-The resulting attribute table will have the same fields of the input layer while
-the features are *aggregated*.
+The resulting attribute table will have the same fields as the input layer
+but the values are not aggregated while the features are *aggregated*.
+Values in the output layer's fields are the ones of the first input feature
+that happens to be processed.
```
Comment (havatv) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: The resulting attribute table will have the same fields as the input layer.
The values in the output layer's fields are the ones of the first input feature
that happens to be processed.

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 4]
```diff
@@ -809,25 +809,24 @@ the features are *aggregated*.
...
+``Input layer`` [vector: any]
+  Vector layer to dissolve.
 
-``Unique ID fields`` [tablefield: any]
+``Dissolve field(s)`` [tablefield: any] [list]
   Optional
 
-  If features share a common value in all selected field(s) their geometries will
-  be combined.
+  Features sharing common value in all selected field(s) will be combined.
```
Comment (havatv) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: Features sharing a common value

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 5]
```diff
@@ -809,25 +809,24 @@ the features are *aggregated*.
...
+  If no field is provided then all the features are dissolved in a single one.
```
Comment (havatv) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: If no field is provided then all the features are dissolved, resulting in a single feature.

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst (hunk 2 above)
Comment (DelazJ) [reply source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: You sure? http://www.wordreference.com/enfr/dissolve

Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: The meaning of dissolve in this context is a bit strange. If something is dissolved here, it is the boundaries.  So I think it is better to describe the result, and not be too concerned with the word dissolve.

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst (hunk 4 above)
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: Better: "Features having the same value for the selected field(s) will be combined."

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst (hunk 2 above)
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: or "in a single one" -> "into a single feature"

**Pull request merged**